internal/rego/rego.go

```diff
@@ -241,9 +248,54 @@ func (r *Rego) parseAnnotations(annotations *ast.Annotations) error {
 		r.enforcement = e
 	}
 
+	metaAnnotations, ok := annotations.Custom[annoAnnotations]
+	if ok {
+		a, ok := metaAnnotations.(map[string]interface{})
+		if !ok {
+			return fmt.Errorf("supplied annotations value is not a map[string]interface{}: %T", metaAnnotations)
+		}
+		if r.metaData == nil {
+			r.metaData = &MetaData{}
+		}
+		ans, err := switchToMap(a)
```

Comment:
I think this is redundant, you can cast to `map[string]string` directly which is the only allowed type for K8s annotations.